src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -59,20 +59,27 @@ public void SerializeUnsigned(BinaryWriter writer)
             }
         }
 
-        private JObject GetInvokeResult(byte[] script, Signers signers = null)
+        private JObject GetInvokeResult(byte[] script, Signers signers = null, bool useDiagnostic = false)
         {
             Transaction tx = signers == null ? null : new Transaction
             {
                 Signers = signers.GetSigners(),
                 Attributes = System.Array.Empty<TransactionAttribute>(),
                 Witnesses = signers.Witnesses,
             };
-            using ApplicationEngine engine = ApplicationEngine.Run(script, system.StoreView, container: tx, settings: system.Settings, gas: settings.MaxGasInvoke);
+            using ApplicationEngine engine = ApplicationEngine.Run(script, system.StoreView, container: tx, settings: system.Settings, gas: settings.MaxGasInvoke, diagnostic: useDiagnostic ? new Diagnostic() : null);
             JObject json = new();
             json["script"] = Convert.ToBase64String(script);
             json["state"] = engine.State;
             json["gasconsumed"] = engine.GasConsumed.ToString();
             json["exception"] = GetExceptionMessage(engine.FaultException);
+            if (useDiagnostic)
+            {
+                json["diagnostics"] = new JObject()
+                {
+                    ["invokedcontracts"] = engine.Diagnostic.InvocationTree.GetItems().Distinct().Select(v => (JString)v.ToString()).ToArray()
```

Comment:
Returns the `InvocationTree`?

Reply:
Also we can return the storage changes?

Reply:
Rename？

Reply:
It is not a simple renaming, but also storing the tree structure into an array.

Reply:
`InvocationTree` seems cannot traverse tree structure outside now. Need some more support.